src/python_testing/TC_CNET_4_12.py

```diff
@@ -0,0 +1,460 @@
+        resp = await self.send_single_cmd(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cmd=cmd
+        )
+        logger.info(f'Step #16: ConnectNetwork resp VARS ({vars(resp)})')
+        logger.info(f'Step #16: ConnectNetwork Status is success ({resp.networkingStatus})')
+        # Verify that the DUT responds with AddThreadNConnectNetworketwork with NetworkingStatus as 'Success'(0)
+        asserts.assert_equal(resp.networkingStatus, Clusters.NetworkCommissioning.Enums.NetworkCommissioningStatusEnum.kSuccess,
+                             "Failure status returned from ConnectNetwork")
+
+        # Wait for the device to establish connection with the new Thread network
+        await asyncio.sleep(60)
+
+        self.step(17)
+        # THREAD_2 - successfully connects to the DUT from previous step
+        networks = await self.read_single_attribute_check_success(
+            cluster=Clusters.NetworkCommissioning,
+            attribute=Clusters.NetworkCommissioning.Attributes.Networks
+        )
+        logger.info(f'Step #17: Networks attribute: {networks}')
+        logger.info(f'Step #17: Networks attribute: {networks.connected}')
+        # Verify that the Thread 2nd is connected
+        asserts.assert_true(network.connected, 'Thread 2nd expected network to be connected, but the connection was not established.')
+
+        self.step(18)
+        breadcrumb_info = await self.read_single_attribute_check_success(
+            cluster=Clusters.GeneralCommissioning,
+            attribute=Clusters.GeneralCommissioning.Attributes.Breadcrumb
+        )
+        logger.info(f'Step #18:  Breadcrumb attribute: {breadcrumb_info}')
+        asserts.assert_equal(breadcrumb_info, 3,
+                             "The Breadcrumb attribute is not 3")
+
+        await asyncio.sleep(60)
+        self.step(19)
+        cmd = Clusters.GeneralCommissioning.Commands.CommissioningComplete()
+        resp = await self.send_single_cmd(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cmd=cmd
+        )
+        logger.info(f'Step #19: CommissioningComplete response ({vars(resp)})')
+        logger.info(f'Step #19: CommissioningComplete Status is success ({resp.errorCode})')
+        # Verify that the DUT responds with CommissioningComplete with ErrorCode as 'OK'(0)
+        asserts.assert_equal(resp.errorCode, Clusters.GeneralCommissioning.Enums.CommissioningErrorEnum.kOk,
+                             "Failure status returned from CommissioningComplete")
+
+        self.step(20)
+        networks = await self.read_single_attribute_check_success(
+            cluster=Clusters.NetworkCommissioning,
+            attribute=Clusters.NetworkCommissioning.Attributes.Networks
+        )
+        logger.info(f'Step #20: Networks attribute: {networks}')
```

Comment:
I think it might make sense to add some steps here to put this device back on the original network. This would look nearly identical to the swap over to network 2, but for network 1.